plugins/redis/plugin.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2018-present the CoreDHCP Authors. All rights reserved
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+package redisplugin
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/coredhcp/coredhcp/handler"
+	"github.com/coredhcp/coredhcp/logger"
+	"github.com/coredhcp/coredhcp/plugins"
+	"github.com/gomodule/redigo/redis"
+	"github.com/insomniacslk/dhcp/dhcpv4"
+	"github.com/insomniacslk/dhcp/dhcpv6"
+)
+
+// various global variables
+var log = logger.GetLogger("plugins/redis")
+var pool *redis.Pool
+var leaseTime time.Duration
+
+func init() {
+	plugins.RegisterPlugin("redis", setupRedis6, setupRedis4)
+}
+
+// Handler6 handles DHCPv6 packets for the redis plugin
+func Handler6(req, resp dhcpv6.DHCPv6) (dhcpv6.DHCPv6, bool) {
+	// TODO add IPv6 support
+	return nil, true
+}
+
+// Handler4 handles DHCPv4 packets for the redis plugin
+func Handler4(req, resp *dhcpv4.DHCPv4) (*dhcpv4.DHCPv4, bool) {
+	// Get redis connection from pool
+	conn := pool.Get()
+
+	defer conn.Close()
+
+	// Get all options for a MAC
+	options, err := redis.StringMap(conn.Do("HGETALL", "mac:"+req.ClientHWAddr.String()))
+	if err == redis.ErrNil || options["ipv4"] == "" {
```

Comment:
When is this `redis.ErrNil` returned?

Reply:
It would be returned when there are no values found for the key.  So basically that is saying if no values are found or there is not an ipv4 address field.

Reply:
Ok, I would add a comment to clarify the `redis.ErrNil` error.